sdk/azidentity/azure_cli_credential_test.go

```diff
@@ -29,6 +29,28 @@ var (
 	}
 )
 
+func TestAzureCLICredential_Error(t *testing.T) {
+	// GetToken shouldn't invoke the CLI a second time after a failure
+	authNs := 0
+	o := AzureCLICredentialOptions{
+		tokenProvider: func(context.Context, string, string) ([]byte, error) {
+			authNs++
+			return nil, errors.New("it didn't work")
+		},
+	}
+	cred, err := NewAzureCLICredential(&o)
+	if err != nil {
+		t.Fatal(err)
```

Comment:
Output that we were expecting a failure but got success

Reply:
This error should be nil because it's from the constructor